packages/backend/src/models/SavedChartModel.ts

```diff
@@ -876,13 +880,28 @@ export class SavedChartModel {
                         `${SpaceTableName}.name as spaceName`,
                         `${PinnedListTableName}.pinned_list_uuid`,
                     ])
-                    .where(
-                        `${SavedChartsTableName}.saved_query_uuid`,
-                        savedChartUuid,
-                    )
                     .orderBy('saved_queries_versions.created_at', 'desc')
                     .limit(1);
 
+                if (isUuid) {
+                    void chartQuery.where((builder) => {
+                        void builder
+                            .where(
+                                `${SavedChartsTableName}.saved_query_uuid`,
+                                savedChartUuidOrSlug,
+                            )
+                            .orWhere(
+                                `${SavedChartsTableName}.slug`,
+                                savedChartUuidOrSlug,
+                            );
+                    });
+                } else {
+                    void chartQuery.where(
+                        `${SavedChartsTableName}.slug`,
+                        savedChartUuidOrSlug,
+                    );
+                }
```

Comment:
[nitpick] wouldn't this be the same as just having:

```suggestion
                void chartQuery.where((builder) => {
                    void builder
                        .where(
                            `${SavedChartsTableName}.saved_query_uuid`,
                            savedChartUuidOrSlug,
                        )
                        .orWhere(
                            `${SavedChartsTableName}.slug`,
                            savedChartUuidOrSlug,
                        );
                });
```

without the need for checking for uuid validity?

Reply:
I wish! The problem is that the database does a UUID validation on filters that are a UUID field. So if we pass `some-dashboard-slug` to the `where uuid =` clause, postgres will throw an type error. We have to only send the value to a UUID field in the DB if we're certain it's that type.